### ISSUE TITLE
[🐛 | Bug]: fusermount: mount failed: Operation not permitted

### ISSUE THREAD

**Issue by FirelightFlagboy** [Scille/parsec-cloud#4760]
### Version

`2.15.1` & `2.16.0-a.0`

### What platforms you were running our software when the bug happened ?

Linux

### What happened ?

I can mount my workspace :cry: on a "clean" install of archlinux.

the relevant part seems to be 

```
fusermount: mount failed: Operation not permitted
```

### Relevant output

```log
fusermount: mount failed: Operation not permitted
2023-06-25T14:38:13.461592Z [error    ] Exception in Qt slot           
Traceback (most recent call last):
  File "/snap/parsec/251/site-packages/parsec/core/mountpoint/fuse_runner.py", line 213, in _run_fuse_thread
    FUSE(
  File "/snap/parsec/251/site-packages/fuse.py", line 711, in __init__
    raise RuntimeError(err)
RuntimeError: 1

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/snap/parsec/251/site-packages/parsec/core/gui/trio_jobs.py", line 123, in __call__
    result = await cast(Awaitable[R], self._fn(*self._args, **self._kwargs))
  File "/snap/parsec/251/site-packages/parsec/core/gui/workspaces_widget.py", line 96, in _do_workspace_mount
    await core.mountpoint_manager.mount_workspace(workspace_id, timestamp)
  File "/snap/parsec/251/site-packages/parsec/core/mountpoint/manager.py", line 243, in mount_workspace
    runner_task = await self._mount_workspace_helper(workspace)
  File "/snap/parsec/251/site-packages/parsec/core/mountpoint/manager.py", line 216, in _mount_workspace_helper
    runner_task = cast(TaskStatus[Path], await start_task(self._nursery, curried_runner))
  File "/snap/parsec/251/site-packages/parsec/utils.py", line 182, in start_task
    return await nursery.start(TaskStatus.wrap_task, corofn, *args, name=name)
  File "/snap/parsec/251/site-packages/parsec/service_nursery.py", line 186, in start
    return await type(nursery).start(nursery, wrap_child, name=name or async_fn)
  File "/snap/parsec/251/site-packages/trio/_core/_run.py", line 1091, in start
    task._eventual_parent_nursery = self
  File "/snap/parsec/251/site-packages/trio/_core/_run.py", line 850, in __aexit__
    raise combined_error_from_nursery
  File "/snap/parsec/251/site-packages/parsec/service_nursery.py", line 184, in wrap_child
    await async_fn(*args, task_status=task_status)
  File "/snap/parsec/251/site-packages/parsec/utils.py", line 166, in wrap_task
    await corofn(*args, task_status=status)
  File "/snap/parsec/251/site-packages/parsec/core/mountpoint/manager.py", line 176, in curried_runner
    async with self._runner(
  File "/snap/parsec/251/usr/lib/python3.9/contextlib.py", line 175, in __aenter__
    return await self.gen.__anext__()
  File "/snap/parsec/251/site-packages/parsec/core/mountpoint/fuse_runner.py", line 243, in fuse_mountpoint_runner
    yield mountpoint_path
  File "/snap/parsec/251/usr/lib/python3.9/contextlib.py", line 193, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/snap/parsec/251/site-packages/parsec/utils.py", line 259, in open_service_nursery
    yield nursery
  File "/snap/parsec/251/usr/lib/python3.9/contextlib.py", line 193, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/snap/parsec/251/site-packages/parsec/service_nursery.py", line 194, in open_service_nursery_with_exception_group
    child_task_scopes.shield = False
  File "/snap/parsec/251/site-packages/trio/_core/_run.py", line 850, in __aexit__
    raise combined_error_from_nursery
  File "/snap/parsec/251/site-packages/parsec/service_nursery.py", line 161, in wrap_child
    await coroutine
  File "/snap/parsec/251/site-packages/trio/_threads.py", line 215, in to_thread_run_sync
    return await trio.lowlevel.wait_task_rescheduled(abort)
  File "/snap/parsec/251/site-packages/trio/_core/_traps.py", line 166, in wait_task_rescheduled
    return (await _async_yield(WaitTaskRescheduled(abort_func))).unwrap()
  File "/snap/parsec/251/site-packages/outcome/_impl.py", line 138, in unwrap
    raise captured_error
  File "/snap/parsec/251/site-packages/trio/_threads.py", line 161, in do_release_then_return_result
    return result.unwrap()
  File "/snap/parsec/251/site-packages/outcome/_impl.py", line 138, in unwrap
    raise captured_error
  File "/snap/parsec/251/site-packages/trio/_threads.py", line 175, in worker_fn
    ret = sync_fn(*args)
  File "/snap/parsec/251/site-packages/parsec/core/mountpoint/fuse_runner.py", line 228, in _run_fuse_thread
    raise MountpointDriverCrash(
parsec.core.mountpoint.exceptions.MountpointDriverCrash: Fuse has crashed on [ABRIDGED]: EPERM

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/snap/parsec/251/site-packages/parsec/core/gui/workspaces_widget.py", line 450, in on_mount_error
    assert timestamp is not None
AssertionError
```

**Comment by FirelightFlagboy** [comment]
After install `fuse3` & `fuse2`, I can mount the workspace :thinking:

**Comment by FirelightFlagboy** [comment]
So after remove the package `fuse2`, the workspaces can't be mounted

**Comment by FirelightFlagboy** [comment]
We install the package [`fuse`]() during the snap packaging.

On ubuntu-20.04 (i.e. core20), it contains those [files](https://packages.ubuntu.com/kinetic/amd64/fuse/filelist)

We should try to copy the binaries in `/bin` & `/sbin`

**Comment by FirelightFlagboy** [comment]
I've tried to include `fusermount` & co from the ubuntu package in the snap without any success, I still get the error :(

I've also tried to set `LD_LIBRARY_PATH` to make sure our lib are loaded first, still no luck.

**Comment by FirelightFlagboy** [comment]
From the suggestion of @vxmichel,

I've installed the package `fuse2` on my system and I'm removing the installed file 1 by 1.

- Removing `/usr/lib/libfuse.so*` cause no problem.
- Removing `/usr/include/{ulockmgr.h,fuse{.h,}}` cause no problem.
- Removing `/usr/lib/libulockmgr.so*` cause no problem.
- Removing `/usr/share/man/man1/fusermount.1.gz /usr/share/man/man1/ulockmgr_server.1.gz /usr/share/man/man8/mount.fuse.8.gz` cause no problem.
- Removing `fusermount` cause a problem:
  
  ```
  fusermount: mount failed: Operation not permitted
  ```

**Comment by FirelightFlagboy** [comment]
Fixed in 35f22c7
